test/functionalTest/cases/3068_non_legacy_notifications/non_legacy_notifications_with_providers.test

```diff
@@ -0,0 +1,395 @@
+# Copyright 2019 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Query forwarding in NGSIv2
+
+--SHELL-INIT--
+dbInit CB
+dbInit CP1
+dbInit CP2
+dbInit CP3
+dbInit CP4
+brokerStart CB 38,186-187,235 IPV4
+brokerStart CP1
+brokerStart CP2
+brokerStart CP3
+brokerStart CP4
+
+--SHELL--
+
+#
+# 01. Register entities of type T, with ID .*, and attribute A1, for CP1
```

Comment:
NGSIv2 based forwarding should cover also not-patter ID registrations.

I'd recomment to have a look to .test in cases/0787_cprs_full_functional_v2/. We did a quite comprehensive testing when we implemented query/update forwarding first time so "porting" these cases (or the relevant sub-sets) to NGSIv2 forwarding would be great.

Reply:
In addition, it would be great to include some basic interoperability case. For instance:

1. Register E1-A1 in NGSIv1-based CPr
3. Register E1-A2 in NGSIv2-based CPr
4. Query GET /v2/entities/E1 and get E-(A1,A2) seamlessly
5. Query POST /v1/queryContext (for E1) and get E-(A1,A2) seamlessly

Reply:
ok. Will copy the query part of that functest and "translate" it into V2

Reply:
Where cases/0787_cprs_full_functional_v2/ finally examined for relevant .test cases to migrate to V2?

Reply:
Just took a look at this ...
There are 24 functest files.
We seems to use V1 registrations in all 24 test files. I assume this (078will 7) is long before we implemented regs in NGSIv2.
What probably should be done here is to remove all usage of V1 regs and only use V2 regs.
V1 is deprecated and will (hopefully) soon be removed.

However, this PR has gone on for quite some time. I'd prefer to postpone this to some other time.
Perhaps to when we finally decide to remove V1. There will be an important clean of functest that day ...

Reply:
Maybe the task about translation case/0787 to NGSIv2 could be done in the future. Probably some of the .test there mix query and update forwards, so it is better to do it at the end. I mean in this order:

* NGSIv2 query forwarding with basic .test (this PR)
* NGSIv2 update forwarding with basic .test (the next PR)
* "Translate" the 0787 cases to NGSIv1 (the next of the next PR)

Makes sense?

Reply:
Sure. Another option is to not modify all those tests but to copy them and convert into pure V2.
Like we've done a few times already.
When we remove all V1, we'll remove the V1 tests as well (they will fail, so no issue/FIXME needed).

Reply:
Right, "copy & translate" instead of "just translate" (case/0787 will remain as it is now, a new case/3068 will be created based on that).

NTC

Reply:
For this, we may need an issue ...
Or, will we remember a few weeks from now, when V2 Update forwarding is ready?

Reply:
It has been noted in https://github.com/telefonicaid/fiware-orion/issues/3068#issuecomment-479933791 so we don't forget about it.

Reply:
I missed this in the long thread of comments above:

> In addition, it would be great to include some basic interoperability case. For instance:
> 
> 1. Register E1-A1 in NGSIv1-based CPr
> 3. Register E1-A2 in NGSIv2-based CPr
> 4. Query GET /v2/entities/E1 and get E-(A1,A2) seamlessly
> 5. Query POST /v1/queryContext (for E1) and get E-(A1,A2) seamlessly

Has been finally included?

If not, I suggest to adding them to this PR, but I'll write them down as pending work (i.e. an extra PR) in #3068

Reply:
Not included. Add to issue

Reply:
Added to the pending-things comment https://github.com/telefonicaid/fiware-orion/issues/3068#issuecomment-479933791

NTC in this PR